### PR TITLE
[SPARK-40823][CONNECT] Connect Proto should carry unparsed identifiers

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Before this PR, connect proto defines the UnresolvedRelation as
```
  message NamedTable {
    repeated string parts = 1;
  }
```
which asked clients to provide multiple name parts of the relation. For example, user could offer `a.b.c.d` as the table name. However, this actually asks clients to implement `CatalystSqlParser.parseMultipartIdentifier`.  The problem to clients is they cannot access the catalyst parser thus needs to re-invent the wheel. Another problem is clients might not be able to implement the parsing correctly.

This PR proposes to change the proto to 
```
  message NamedTable {
    string unparsed_identifier = 1;
}
```
which only needs clients to provide the user's table name. Server side as it can access catalyst, can parse the identifier.

This PR also changes on the Column identifier accordingly.

### Why are the changes needed?

This proposal reduced the required work on the client side.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def __init__(self, table_name: str) -> None:
 
     def plan(self, session: Optional["RemoteSparkSession"]) -> proto.Relation:
         plan = proto.Relation()
-        plan.read.named_table.parts.extend(self.table_name.split("."))
+        plan.read.named_table.unparsed_identifier = self.table_name
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/plan.py:111]: I'm not sure this is the only place where this is used like that, I know this is the lack of test coverage, but I'd suggest to search for `split(".")`.

Comment (grundprinzip) [reply python/pyspark/sql/connect/plan.py:111]:
```❯ rg 'split\(' python/pyspark/sql/connect/
python/pyspark/sql/connect/plan.py
111:        plan.read.named_table.parts.extend(self.table_name.split("."))
160:            else self.unresolved_attr(*(c.split(".")))

python/pyspark/sql/connect/column.py
98:        return ColumnRef(*name.split("."))
```

**Comment by amaliujia** [comment]
R: @cloud-fan @HyukjinKwon

**Review by HyukjinKwon — COMMENTED** [review]
Yeah, +1 for the idea. I don't think we can address pasing the parts within client side.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by zhengruifeng** [comment]
LATE LGTM

**Pull request closed without merging**